kinto/core/resource/__init__.py

```diff
@@ -542,7 +542,8 @@ def patch(self):
                 new_record[extra_field] = existing[extra_field]
 
         # Adjust response according to ``Response-Behavior`` header
-        body_behavior = self.request.headers.get('Response-Behavior', 'full')
+        body_behavior = self.request.validated.get('header',
+                                                   {}).get('Response-Behavior', 'full')
```

Comment:
When does it happen that `validated` doesn't have a `header` entry? See https://github.com/Cornices/cornice/blob/9b73c5ae8dfbebede6413a007ffac7fe28e76401/cornice/validators/__init__.py#L60-L62